sdks/go/client.go

```diff
@@ -0,0 +1,221 @@
+package hatchet
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v0Client "github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+	"github.com/hatchet-dev/hatchet/sdks/go/features"
+)
+
+// Client provides the main interface for interacting with Hatchet.
+type Client struct {
+	v0Client v0Client.Client
+}
+
+// NewClient creates a new Hatchet client.
+// Configuration options can be provided to customize the client behavior.
+func NewClient(opts ...v0Client.ClientOpt) (*Client, error) {
+	v0Client, err := v0Client.New(opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		v0Client: v0Client,
+	}, nil
+}
+
+// NewWorker creates a worker that can execute workflows.
+// The worker is configured using functional options.
+func (c *Client) NewWorker(name string, options ...WorkerOption) (*worker.Worker, error) {
+	config := &workerConfig{
+		slots:        100,
+		durableSlots: 1000,
+	}
+
+	for _, opt := range options {
+		opt(config)
+	}
+
+	// Use the higher of slots or durableSlots as the max runs
+	// since durable tasks tend to be longer-running
+	maxRuns := config.slots
+	if config.durableSlots > maxRuns {
+		maxRuns = config.durableSlots
+	}
+
+	workerOpts := []worker.WorkerOpt{
+		worker.WithClient(c.v0Client),
+		worker.WithName(name),
+		worker.WithMaxRuns(maxRuns),
+	}
```

Comment:
this is a bit confusing, but in Python / TS we're basically running two workers side by side - one for durable tasks and another for non-durable tasks. I think we need to do the same here. we basically have two entirely separate action registries, etc., and then each one gets assigned the proper number of slots as is passed in